src/App.tsx

```diff
@@ -15,36 +15,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = 'default',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const preparedGoods = [...goodsFromServer];
+
+const sortGoods = (order: SortType, isReversed: boolean): string[] => {
+  let sortedGoods: string[];
+
+  switch (order) {
+    case SortType.ALPHABET:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      sortedGoods = [...preparedGoods].sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...preparedGoods];
+  }
+
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>(preparedGoods);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortOrder, setSortOrder] = useState<SortType>(SortType.DEFAULT);
+
+  const handleSort = (order: SortType) => {
+    const sortedGoods = sortGoods(order, isReversed);
+
+    setGoods(sortedGoods);
+    setSortOrder(order);
+  };
+
+  const reverseGoods = () => {
+    setGoods(prevGoods => [...prevGoods].reverse());
+    setIsReversed(prevState => !prevState);
+  };
+
+  const resetGoods = () => {
+    setGoods(preparedGoods);
+    setSortOrder(SortType.DEFAULT);
+    setIsReversed(false);
+  };
+
+  const isResetVisible = sortOrder !== SortType.DEFAULT || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${sortOrder !== SortType.ALPHABET ? 'is-light' : ''}`}
```

Comment:
classnames will be a better option (+ 'is-' classes will conflic with eachoter )